spras/config.py

```diff
@@ -252,11 +257,27 @@ def process_config(self, raw_config):
         self.analysis_include_ml = raw_config["analysis"]["ml"]["include"]
         self.analysis_include_evaluation = raw_config["analysis"]["evaluation"]["include"]
 
+        # Only run ML aggregate per algorithm if analysis include ML is set to True
+        if 'aggregate_per_algorithm' in self.ml_params and self.analysis_include_ml:
+            self.analysis_include_ml_aggregate_algo = raw_config["analysis"]["ml"]["aggregate_per_algorithm"]
+        else:
+            self.analysis_include_ml_aggregate_algo = False
+
+        # Raises an error if Evaluation is enabled but no gold standard data is provided
         if self.gold_standards == {} and self.analysis_include_evaluation:
             raise ValueError("Evaluation analysis cannot run as gold standard data not provided. "
                              "Please set evaluation include to false or provide gold standard data.")
 
-        if 'aggregate_per_algorithm' in self.ml_params and self.analysis_include_ml:
-            self.analysis_include_ml_aggregate_algo = raw_config["analysis"]["ml"]["aggregate_per_algorithm"]
+         # Only run Evaluation if ML is set to True
+        if not self.analysis_include_ml and self.analysis_include_evaluation:
+            self.analysis_include_evaluation = False
+
+        # Only run Evaluation aggregate per algorithm if analysis include ML is set to True
+        if 'aggregate_per_algorithm' in self.evaluation_params and self.analysis_include_evaluation:
+            self.analysis_include_evaluation_aggregate_algo = raw_config["analysis"]["evaluation"]["aggregate_per_algorithm"]
         else:
-            self.analysis_include_ml_aggregate_algo = False
+            self.analysis_include_evaluation_aggregate_algo = False
+
+        # Only run Evaluation per algorithm if ML per algorithm is set to True
+        if not self.analysis_include_ml_aggregate_algo and self.analysis_include_evaluation_aggregate_algo:
```

Comment:
Same question for `self.analysis_include_evaluation_aggregate_algo` here

Reply:
I was being very explicit about the use case when I made this. We don't actually need self.analysis_include_evaluation_aggregate_algo.